admin_guide/high_availability.adoc

```diff
@@ -11,37 +11,54 @@
+reachable the VIP will not be assigned to the node. When the port is set to "0" the check is suppressed.
 
-Use cases for high-availability include:
+IP Failover uses link:http://www.keepalived.org/[*Keepalived*] to host a set of externally accessible
+Virtual IP (VIP) addresses on a set of hosts.  Each VIP is serviced by, at most, one of the hosts at
+any point in time.  *Keepalived* uses the VRRP protocol to determine which host in the set of hosts
+will service each VIP address.  If a host goes down or the service that *Keepalived* is watching goes down the
+VIP is switched to another host in the set. As long as a single host is available the VIP is serviced.
 
-* I want my cluster to be assigned a resource set and I want the cluster to automatically manage those resources.
-* I want my cluster to be assigned a set of VIPs that the cluster manages and migrates (with zero or minimal downtime) on failure conditions, and I should not be required to perform any manual interactions to update the upstream "discovery" sources (e.g., DNS). The cluster should service all the assigned VIPs when at least a single node is available, despite the current available resources not being sufficient to reach the desired state.
+{product-title} supports an ipfailover Deployment Config created by the `oadm ipfailover` command. The ipfailover
+Deployment Config specifies the set of VIP addresses and the set of nodes on which to service them. The cluster
+can have multiple ipfailover Deployment Configs, each servicing its own set of unique VIP addresses. Each pod in the
+ipfailover Deployment Config runs *Keepalived* and each pod runs on a different node in the set of ipfailover nodes.
 
+When using VIPs to access a pod with host networking, e.g. a router, the application pod should be running on all of
+ the nodes that are running the ipfailover pods. This permits any of the ipfailover nodes to become the master and
+service the VIP.  A mismatch will result in either some ipfailover nodes never servicing the VIPs or some application
+pods to never receive traffic.  Using the same selector and replication count for both ipfailover and the appliction
+will eliminate the mismatch.
+
+When using VIPs to access a service any of the nodes can be in the ipfailover set of nodes, since the service is
+reachable on all nodes (no matter where the application pod really is running).  Any of the ipfailover nodes can
+be master at any time. The service must configure the externalIPs to be the set of VIPs or the service can use a
+NodePort and use any VIPs that route to the nodes. The ipfailover watch port can be set to the service port.
+
+IMPORTANT: Even though a service VIP is highly available, performance can still be affected. *keepalived*
+makes sure that each of the VIPs is served by some node in the configuration, and several VIPs may end up
+on the same node even when other nodes have none. Also, high availability does not do load balancing.
```

Comment:
Also, high availability does not do load balancing.

High availability does not inherently do load balancing.  But you can expose multiple IPs for the same resource and load balance them outside the cluster, or serve all of the IPs in DNS.